main.go

```diff
@@ -102,15 +102,20 @@ func main() {
 	if shouldUpdate {
 		err = ctrlClient.Update(ctx, &node)
 		if err != nil {
-			fmt.Printf("ERROR: failed to apply new labels")
+			fmt.Printf("ERROR: failed to apply new labels\n")
 			panic(err)
 		}
-		fmt.Printf("new labels applied to node")
+		fmt.Printf("new labels applied to node\n")
 	} else {
-		fmt.Printf("required labels are already applied")
+		fmt.Printf("required labels are already applied\n")
 	}
 
-	fmt.Printf("capi-node-labeler finished successfully")
+	fmt.Printf("capi-node-labeler finished successfully\n")
+	fmt.Printf("sleeping forever\n")
+
+	select {
+	// sleeping forever
+	}
```

Comment:
unfortunately no, DaemonSet can have only `RestartPolicy: Always` which means if the pod finished it would be started again by Kubernetes

I tried that before and hoped that I can be done by settings RestartPolicy to `Never` or `OnFailure` but that did not worked as that value is not allowed for DS :(

Reply:
Oh right. Ok then, thanks for explaining